app/src/main/java/org/stepic/droid/ui/activities/AnimatedOnboardingActivity.kt

```diff
@@ -0,0 +1,114 @@
+package org.stepic.droid.ui.activities
+
+import android.content.pm.ActivityInfo
+import android.os.Bundle
+import android.support.v4.view.ViewPager
+import kotlinx.android.synthetic.main.activity_onboarding.*
+import org.stepic.droid.R
+import org.stepic.droid.analytic.Analytic
+import org.stepic.droid.base.FragmentActivityBase
+import org.stepic.droid.ui.activities.contracts.OnNextClickedListener
+import org.stepic.droid.ui.adapters.OnboardingAdapter
+import org.stepic.droid.ui.custom.OnboardingPageTransformer
+import org.stepic.droid.ui.fragments.OnboardingFragment
+import timber.log.Timber
+
+
+class AnimatedOnboardingActivity : FragmentActivityBase(), OnNextClickedListener {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
+        setContentView(R.layout.activity_onboarding)
+        initViewPager()
+        initClose()
+    }
+
+    private fun initViewPager() {
+        onboardingViewPager.adapter = OnboardingAdapter(supportFragmentManager)
+        onboardingCircleIndicator.setViewPager(onboardingViewPager)
+        val pageChangeListener: ViewPager.OnPageChangeListener = object : ViewPager.OnPageChangeListener {
+            override fun onPageScrollStateChanged(state: Int) {
+            }
+
+            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+            }
+
+            override fun onPageSelected(position: Int) {
+                invokeAnimationOnFragment(position)
+                reportToAnalytic(Analytic.Onboarding.SCREEN_OPENED)
+            }
+        }
+        onboardingViewPager.addOnPageChangeListener(pageChangeListener)
+        onboardingViewPager.setPageTransformer(false, OnboardingPageTransformer())
```

Comment:
Maybe we should set 
```kotlin
onboardingViewPager.offscreenPageLimit = onboardingViewPager.adapter.count
```
to avoid page loading during swipes.